client/data/stats/use-has-never-published-post.ts

```diff
@@ -0,0 +1,38 @@
+import { useQuery } from 'react-query';
+import wpcom from 'calypso/lib/wp';
+import type { QueryOptions, UseQueryResult } from 'react-query';
+
+export const useHasNeverPublishedPostCacheKey = (
+	siteId: number | null,
+	includePages: boolean
+) => [ 'site-has-never-published-post', siteId, includePages ];
+
+interface Options extends QueryOptions {
+	enabled?: boolean;
+}
+
+export const useHasNeverPublishedPost = (
+	siteId: number | null,
+	includePages: boolean,
+	queryOptions: Options
+): UseQueryResult< boolean > => {
+	const { enabled = true } = queryOptions;
+	return useQuery(
+		useHasNeverPublishedPostCacheKey( siteId, includePages ),
+		() => fetchHasNeverPublishedPost( siteId, includePages ),
+		{
+			...queryOptions,
+			enabled: !! siteId && enabled,
+		}
+	);
+};
+
+function fetchHasNeverPublishedPost(
+	siteId: number | null,
+	includePages: boolean
+): Promise< unknown > {
```

Comment:
Shouldn't we define a type rather than using `unknown`?

```suggestion
): Promise< boolean > {
```